service/history/shard/controller.go

```diff
@@ -163,7 +165,7 @@ func (c *controller) Start() {
 	c.shutdownWG.Add(1)
 	go c.shardManagementPump()
 
-	err := c.GetHistoryServiceResolver().AddListener(shardControllerMembershipUpdateListenerName, c.membershipUpdateCh)
+	err := c.GetMembershipResolver().Subscribe(service.History, shardControllerMembershipUpdateListenerName, c.membershipUpdateCh)
 	if err != nil {
 		c.logger.Error("Error adding listener", tag.Error(err))
```

Comment:
Please update this error message, as this is not listener anymore, but a subscriber.